third_party/nvfuser/csrc/executor.cpp

```diff
@@ -797,6 +797,15 @@ FusionExecutor::GlobalBuffers FusionExecutor::allocGlobalVals(
   return global_buffers;
 }
 
+std::vector<at::Tensor> FusionExecutor::allocOutputSpace(
+    const at::ArrayRef<IValue>& inputs) {
+  auto kernel_inputs = KernelArgumentHolder::createKernelArgumentHolder(inputs);
+  auto expr_eval =
+      executor_utils::bindInputs(kernel_inputs, lowered_->kernel());
+  auto outputs = allocOutputs(kernel_inputs, expr_eval);
+  return outputs;
+}
```

Comment:
Is this needed?

Reply:
Allocating the output is not always needed -- only if we actually receive this tensor from a Send/Recv later on.
However, what is always needed is to infer the dimensions of the output, and this is one way to do it. But you are right it probably could be refined